test_conformance/images/clCopyImage/test_copy_1D_buffer.cpp

```diff
@@ -0,0 +1,516 @@
+                } while ((imageInfo.rowPitch % pixelSize) != 0);
+            }
+
+            log_info("Testing %d\n", (int)sizes[idx][0]);
+            if (gDebugTrace)
+                log_info("   at max size %d\n", (int)sizes[idx][0]);
+            if (test_copy_image_size_1D_buffer(context, queue, &imageInfo,
+                                               &imageInfo, seed))
+                return -1;
+        }
+    }
+    else
+    {
+        for (int i = 0; i < NUM_IMAGE_ITERATIONS; i++)
+        {
+            cl_ulong size;
+            size_t rowPadding = gEnablePitch ? 48 : 0;
+            // Loop until we get a size that a) will fit in the max alloc size
+            // and b) that an allocation of that image, the result array, plus
+            // offset arrays, will fit in the global ram space
+            do
+            {
+                imageInfo.width =
+                    (size_t)random_log_in_range(16, (int)maxWidth / 32, seed);
+
+                imageInfo.rowPitch = imageInfo.width * pixelSize + rowPadding;
+
+                if (gEnablePitch)
+                {
+                    do
+                    {
+                        rowPadding++;
+                        imageInfo.rowPitch =
+                            imageInfo.width * pixelSize + rowPadding;
+                    } while ((imageInfo.rowPitch % pixelSize) != 0);
+                }
+
+                size = (size_t)imageInfo.rowPitch * 4;
+            } while (size > maxAllocSize || (size * 3) > memSize);
+
+            if (gDebugTrace)
+            {
+                log_info("   at size %d (row pitch %d) out of %d\n",
+                         (int)imageInfo.width, (int)imageInfo.rowPitch,
+                         (int)maxWidth);
+            }
+
+            int ret = test_copy_image_size_1D_buffer(context, queue, &imageInfo,
+                                                     &imageInfo, seed);
+            if (ret) return -1;
+        }
+    }
+
+    return 0;
+}
+
+int test_copy_image_set_1D_1D_buffer(cl_device_id device, cl_context context,
+                                     cl_command_queue queue,
+                                     cl_image_format *format)
+{
+    size_t maxWidth;
+    cl_ulong maxAllocSize, memSize;
+    image_descriptor imageInfo = { 0 };
+    RandomSeed seed(gRandomSeed);
+    size_t pixelSize;
+
+    if (gTestMipmaps)
+    {
+        // 1D image buffers don't support mipmaps
+        // https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_Ext.html#cl_khr_mipmap_image
+        return 0;
+    }
+
+    imageInfo.format = format;
+    imageInfo.height = imageInfo.depth = imageInfo.arraySize =
+        imageInfo.slicePitch = 0;
+    imageInfo.type = CL_MEM_OBJECT_IMAGE1D_BUFFER;
+    pixelSize = get_pixel_size(imageInfo.format);
+
+    int error = clGetDeviceInfo(device, CL_DEVICE_IMAGE2D_MAX_WIDTH,
+                                sizeof(maxWidth), &maxWidth, NULL);
```

Comment:
Can you check if this is correct?  It's not clear to me if this value is used to create a 1D image, or a 1D image from a buffer, or both.  If it's used to create a 1D image from a buffer then should we be querying [CL_DEVICE_IMAGE_MAX_BUFFER_SIZE](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_DEVICE_IMAGE_MAX_BUFFER_SIZE) instead?  Or, if it's both, should we be querying both and using the smallest value?

Reply:
It is used to create 1D images. We could query both to get the smallest, but it would mean we expect to have an implementation with 1D image from buffer smaller than standard 1D image. Is that really possible?